docs: add `resolve` field of Rsbuild config (#517)
web-infra-dev/rslib: website/docs/en/config/rsbuild/resolve.mdx

import { RsbuildDocBadge } from '@components/RsbuildDocBadge';

# resolve

## resolve.aliasStrategy <RsbuildDocBadge path="/config/resolve/alias-strategy" text="resolve.aliasStrategy" />

`resolve.aliasStrategy` is used to control the priority between the `paths` option in `tsconfig.json` and the `alias` option in the bundler.

## resolve.alias <RsbuildDocBadge path="/config/resolve/alias" text="resolve.alias" />

Create aliases to import or require certain modules, same as the [resolve.alias](https://rspack.dev/config/resolve#resolvealias) config of Rspack.

It is important to note that `resolve.alias` differs from [output.externals](/config/rsbuild/output#outputexternals) in the following ways:

> - `resolve.alias` is used when bundling target code into the product (when [lib.bundle](/config/lib/bundle) is set to `true`). It allows you to replace the target module you want to include in the output with another module.

Reviewer: product -> output / bundles

```suggestion
- `resolve.alias` allows you to replace the target module you want to include in the output bundles with another module. It only works if [lib.bundle](/config/lib/bundle) is set to `true`.
```
